sw/airborne/subsystems/gps/gps_datalink.c

```diff
@@ -77,6 +78,10 @@ static void send_magnetometer(int32_t course, uint32_t now_ts)
   mag_real.z = 0;
   MAGS_BFP_OF_REAL(mag, mag_real);
 
+  // update IMU information
+  VECT3_COPY(imu.mag_unscaled, mag);
+  imu_scale_mag(&imu);
```

Comment:
If you call the `imu_scale_mag`, the coefficients from the airframe will be applied. So in this case, you should have a sensitivity of 1 and an offset of 0. If you are fine with that I can validate this code. Otherwise, you can make a direct copy to `imu.mag` as I mentioned before, so airframe calibration has no effect. It is up to you, I don't have a strong preference for this.